### DISCUSSION TITLE
How to resolve $ref from custom data store like AWS DynamoDB

### DISCUSSION THREAD

**Discussion by thulaban1990** [networknt/json-schema-validator#1096] [Q&A] [answered]
Hi, We use this package to validate JSON data with schema. We are storing JSON schemas in a AWS DynamodDB table as we have many of config like this. We have schemas created in a way that it promotes reusability where any schema could be re-used, its created as its own and stored in AWS DynamodDB and these are referenced $ref in other schemas where reusability is needed.

When getting schema to validate main schema, how to resolve and get schema from AWS DynamodDB while resolving $ref from main schema to get other reusable schema based on $ref.

If there is any example on how to fetch schema from custom data store helps us.
Looking forward to hear from you.

**Comment by justin-tay** [comment, marked as answer]
You will just need to implement a custom [SchemaLoader](https://github.com/networknt/json-schema-validator/blob/12807f33a19a7981452b8eabf9e1bcdae30df5c6/src/main/java/com/networknt/schema/resource/SchemaLoader.java#L24) and register that with the factory, see [Customizing Schema Retrieval](https://github.com/networknt/json-schema-validator/blob/master/doc/schema-retrieval.md).

You can refer to [UriSchemaLoader](https://github.com/networknt/json-schema-validator/blob/12807f33a19a7981452b8eabf9e1bcdae30df5c6/src/main/java/com/networknt/schema/resource/UriSchemaLoader.java) as an example
